src/tabpane/TabButton.ts

```diff
@@ -0,0 +1,167 @@
+ */
+export interface TabButtonProperties extends ThemeableProperties {
+	active?: boolean;
+	closeable?: boolean;
+	controls: string;
+	disabled?: boolean;
+	id: string;
+	index: number;
+	onClick?: (index: number) => void;
+	onCloseClick?: (index: number) => void;
+	onEndPress?: () => void;
+	onHomePress?: () => void;
+	onDownArrowPress?: () => void;
+	onLeftArrowPress?: () => void;
+	onRightArrowPress?: () => void;
+	onUpArrowPress?: () => void;
+};
+
+export const TabButtonBase = ThemeableMixin(WidgetBase);
+
+@theme(css)
+export default class TabButton extends TabButtonBase<TabButtonProperties> {
+	private _onClick() {
+		const {
+			disabled,
+			index,
+			onClick
+		} = this.properties;
+
+		!disabled && onClick && onClick(index);
+	}
+
+	private _onCloseClick(event: MouseEvent) {
+		const {
+			index,
+			onCloseClick
+		} = this.properties;
+
+		event.stopPropagation();
+		onCloseClick && onCloseClick(index);
+	}
+
+	private _onKeyDown(event: KeyboardEvent) {
+		const {
+			closeable,
+			disabled,
+			index,
+			onCloseClick,
+			onEndPress,
+			onHomePress,
+			onDownArrowPress,
+			onLeftArrowPress,
+			onRightArrowPress,
+			onUpArrowPress
+		} = this.properties;
+
+		if (disabled) {
+			return;
+		}
+
+		// Accessibility
+		switch (event.keyCode) {
+			// Escape
+			case 27:
+				closeable && onCloseClick && onCloseClick(index);
+				break;
+			// Left arrow
+			case 37:
+				onLeftArrowPress && onLeftArrowPress();
+				break;
+			// Right arrow
+			case 39:
+				onRightArrowPress && onRightArrowPress();
+				break;
+			// Up arrow
+			case 38:
+				onUpArrowPress && onUpArrowPress();
+				break;
+			// Down arrow
+			case 40:
+				onDownArrowPress && onDownArrowPress();
+				break;
+			// Home
+			case 36:
+				onHomePress && onHomePress();
+				break;
+			// End
+			case 35:
+				onEndPress && onEndPress();
+				break;
+		}
+	}
+
+	private _restoreFocus(element: HTMLElement) {
+		this.properties.active && element.focus();
+	}
+
+	protected onElementCreated(element: HTMLElement, key: string) {
+		key === 'tab-button' && this._restoreFocus(element);
+	}
+
+	protected onElementUpdated(element: HTMLElement, key: string) {
+		key === 'tab-button' && this._restoreFocus(element);
+	}
+
+	render(): DNode {
+		const {
+			active,
+			closeable,
+			controls,
+			disabled,
+			id
+		} = this.properties;
+
+		const children = closeable ? this.children.concat([
+			v('button', {
+				tabIndex: active ? 0 : -1,
+				classes: this.classes(css.close),
+				innerHTML: 'close tab',
+				onclick: this._onCloseClick
+			})
+		]) : this.children;
+
+		return v('div', {
+			'aria-controls': controls,
+			'aria-disabled': disabled ? 'true' : 'false',
+			'aria-selected': active ? 'true' : 'false',
+			classes: this.classes(
+				css.tabButton,
+				active ? css.activeTabButton : null,
+				disabled ? css.disabledTabButton : null
+			),
+			id,
+			key: 'tab-button',
+			role: 'tab',
+			tabIndex: active ? 0 : -1,
+			onclick: this._onClick,
```

Comment:
All of widgets uses underscores for any private member. I started doing this after some past PR comments saying we were going to be consistent about doing this.